src/cargo/core/compiler/context/compilation_files.rs

```diff
@@ -421,9 +421,16 @@ impl<'a, 'cfg: 'a> CompilationFiles<'a, 'cfg> {
                 Some(types) => {
                     for file_type in types {
                         let path = out_dir.join(file_type.filename(&file_stem));
-                        let hardlink = link_stem
-                            .as_ref()
-                            .map(|&(ref ld, ref ls)| ld.join(file_type.filename(ls)));
+                        // Don't create hardlink for library tests and integration tests
+                        let hardlink = if unit.target.is_test()
+                            || (unit.target.is_lib() && crate_type == "bin")
```

Comment:
I think you may be able to check `unit.profile.test` perhaps rather than checking for `is_test` and such? That should be whether or not we pass `--test` to rustc.

Reply:
I think it should be:

```rust
let hardlink = if unit.mode.is_any_test() {
```

`test` hasn't been in the profile for a few years. 😜

Reply:
Er my bad, I'm so dated 👴

Reply:
Ok, let me try